Lock wait timeout exceeded

From time to time when I save a job, error `Lock wait timeout exceeded; try restarting transaction.` appears. Only restart of Rundeck helps to resolve the issue. A similar problem was fixed by #4169 but the fix was merged to branch release-2.11

**My Rundeck detail**
* Rundeck version: 3.0.8-20181029
* install type: Docker(deb)
* OS Name/version: ubuntu 16.04
* DB Type/version: mysql

**To Reproduce**
1. Go to 'edit job'
2. Click on 'save'
4. See error

**Desktop**
 - OS: ubuntu 18.10
 - Browser chrome
 - Version 70

<details>
  <summary>Stacktrace</summary>
  2018-11-28 12:13:11.238 ERROR --- [p473344802-1251] o.g.web.errors.GrailsExceptionResolver   : MySQLTransactionRollbackException occurred when processing request: [POST] /project/spark/job/update                                                                                                                                                      [90/1874]
Lock wait timeout exceeded; try restarting transaction. Stacktrace follows:                                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                                                                                                  
org.springframework.dao.CannotAcquireLockException: Hibernate operation: could not execute statement; SQL [n/a]; Lock wait timeout exceeded; try restarting transaction; nested exception is com.mysql.jdbc.exceptions.jdbc4.MySQLTransactionRollbackException: Lock wait timeout exceeded; try restarting transaction                                            
        at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:259)                                                                                                                                                                                                                           
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73)                                                                                                                                                                                                                      
        at org.grails.orm.hibernate.GrailsHibernateTemplate.convertJdbcAccessException(GrailsHibernateTemplate.java:731)                                                                                                                                                                                                                                          
        at org.grails.orm.hibernate.GrailsHibernateTemplate.convertHibernateAccessException(GrailsHibernateTemplate.java:719)                                                                                                                                                                                                                                     
        at org.grails.orm.hibernate.GrailsHibernateTemplate.doExecute(GrailsHibernateTemplate.java:303)                                                                                                                                                                                                                                                           
        at org.grails.orm.hibernate.GrailsHibernateTemplate.execute(GrailsHibernateTemplate.java:243)                                                                                                                                                                                                                                                             
        at org.grails.orm.hibernate.GrailsHibernateTemplate.execute(GrailsHibernateTemplate.java:117)                                                                                                                                                                                                                                                             
        at org.grails.orm.hibernate.AbstractHibernateGormInstanceApi.performSave(AbstractHibernateGormInstanceApi.groovy:246)                                                                                                                                                                                                                                     
        at org.grails.orm.hibernate.AbstractHibernateGormInstanceApi.save(AbstractHibernateGormInstanceApi.groovy:163)                                                                                                                                                                                                                                            
        at org.grails.datastore.gorm.GormEntity$Trait$Helper.save(GormEntity.groovy:151)                                                                                                                                                                                                                                                                          
        at rundeck.services.ScheduledExecutionService.$tt___doupdate(ScheduledExecutionService.groovy:2476)                                                                                                                                                                                                                                                       
        at grails.gorm.transactions.GrailsTransactionTemplate$2.doInTransaction(GrailsTransactionTemplate.groovy:94)                                                                                                                                                                                                                                              
        at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:133)                                                                                                                                                                                                                                                      
        at grails.gorm.transactions.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:91)                                                                                                                                                                                                                                                        
        at rundeck.controllers.ScheduledExecutionController$_update_closure65.doCall(ScheduledExecutionController.groovy:2082)                                                                                                                                                                                                                                    
        at grails.artefact.Controller$Trait$Helper.withForm(Controller.groovy:272)                                                                                                                                                                                                                                                                                
        at grails.artefact.Controller$Trait$Helper.withForm(Controller.groovy:250)                                                                                                                                                                                                                                                                                
        at rundeck.controllers.ScheduledExecutionController.update(ScheduledExecutionController.groovy:2069)                                                                                                                                                                                                                                                      
        at org.grails.core.DefaultGrailsControllerClass$MethodHandleInvoker.invoke(DefaultGrailsControllerClass.java:223)                                                                                                                                                                                                                                         
        at org.grails.core.DefaultGrailsControllerClass.invoke(DefaultGrailsControllerClass.java:188)                                                                                                                                                                                                                                                             
        at org.grails.web.mapping.mvc.UrlMappingsInfoHandlerAdapter.handle(UrlMappingsInfoHandlerAdapter.groovy:90)                                                                                                                                                                                                                                               
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)                                                                                                                                                                                                                                                               
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)                                                                                                                                                                                                                                                                
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)                                                                                                                                                                                                                                                             
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)                                                                                                                                                                                                                                                                     
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)                                                                                                                                                                                                                                                                    
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)                                                                                                                                                                                                                                                                                 
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)                                                                                                                                                                                                                                                                
        at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:215)                                                                                                                                                                                                                                                    
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)                                                                                                                                                                                                                                                                
        at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter$1.run(JaasApiIntegrationFilter.java:96)                                                                                                                                                                                                                                              
        at javax.security.auth.Subject.doAs(Subject.java:422)                                                                                                                                                                                                                                                                                                     
        at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter.doFilter(JaasApiIntegrationFilter.java:105)                                                                                                                                                                                                                                          
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)                                                                                                                                                                                                                                                                
        at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55)                                                                                                                                                                                                                            
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)                                                                                                                                                                                                                                                                
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)                                                                                                                                                                                                                                                                
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317)                                                                                                                                                                                                                                               
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)                                                                                                                                                                                                                                 
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)                                                                                                                                                                                                                                
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)                                                                                                                                                                                                                                       
        at grails.plugin.springsecurity.web.UpdateRequestContextHolderExceptionTranslationFilter.doFilter(UpdateRequestContextHolderExceptionTranslationFilter.groovy:64)                                                                                                                                                                                         
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at grails.plugin.springsecurity.web.filter.GrailsAnonymousAuthenticationFilter.doFilter(GrailsAnonymousAuthenticationFilter.groovy:54)                                                                                                                                                                                                                    
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:158)                                                                                                                                                                                                            
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)                                                                                                                                                                                                         
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter$1.run(JaasApiIntegrationFilter.java:96)                                                                                                                                                                                                                                              
        at javax.security.auth.Subject.doAs(Subject.java:422)                                                                                                                                                                                                                                                                                                     
        at org.springframework.security.web.jaasapi.JaasApiIntegrationFilter.doFilter(JaasApiIntegrationFilter.java:105)                                                                                                                                                                                                                                          
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)                                                                                                                                                                                                       
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.groovy:64)                                                                                                                                                                                                                                     
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)                                                                                                                                                                                                                          
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at grails.plugin.springsecurity.web.SecurityRequestHolderFilter.doFilter(SecurityRequestHolderFilter.groovy:58)                                                                                                                                                                                                                                           
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)                                                                                                                                                                                                                                               
        at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)                                                                                                                                                                                                                                                          
        at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)                                                                                                                                                                                                                                                                  
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)                                                                                                                                                                                                                                                                
        at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:77)                                                                                                                                                                                                                                                     
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1317)                                                                                                                                                                                                                                                                     
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)                                                                                                                                                                                                                                                                       
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)                                                                                                                                                                                                                                                                              
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)                                                                                                                                                                                                                                                                      
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)                                                                                                                                                                                                                                                                       
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1219)                                                                                                                                                                                                                                                                      
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)                                                                                                                                                                                                                                                                          
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)                                                                                                                                                                                                                                                                        
        at org.eclipse.jetty.server.Server.handle(Server.java:531)                                                                                                                                                                                                                                                                                                
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)                                                                                                                                                                                                                                                                                      
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)                                                                                                                                                                                                                                                                            
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)                                                                                                                                                                                                                                                            
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)                                                                                                                                                                                                                                                                                      
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)                                                                                                                                                                                                                                                                                   
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)                                                                                                                                                                                                                                                                 
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)                                                                                                                                                                                                                                                               
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)                                                                                                                                                                                                                                                              
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)                                                                                                                                                                                                                                                                     
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)                                                                                                                                                                                                                                               
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)                                                                                                                                                                                                                                                                       
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)                                                                                                                                                                                                                                                                        
        at java.lang.Thread.run(Thread.java:748)                                                                                                                                                                                                                                                                                                                  
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLTransactionRollbackException: Lock wait timeout exceeded; try restarting transaction                                                                                                                                                                                                                              
        at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)                                                                                                                                                                                                                                                                                                   
        at com.mysql.jdbc.Util.getInstance(Util.java:408)                                                                                                                                                                                                                                                                                                         
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:951)                                                                                                                                                                                                                                                                                          
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)                                                                                                                                                                                                                                                                                             
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)                                                                                                                                                                                                                                                                                             
        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)                                                                                                                                                                                                                                                                                                  
        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)                                                                                                                                                                                                                                                                                               
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2490)                                                                                                                                                                                                                                                                                        
        at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1858)                                                                                                                                                                                                                                                                          
        at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2079)                                                                                                                                                                                                                                                                    
        at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2013)                                                                                                                                                                                                                                                                    
        at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5104)                                                                                                                                                                                                                                                                       
        at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:1998)                                                                                                                                                                                                                                                                            
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:204)                                                                                                                                                                                                                                                     
        at org.hibernate.engine.jdbc.batch.internal.NonBatchingBatch.addToBatch(NonBatchingBatch.java:45)                                                                                                                                                                                                                                                         
        at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3184)                                                                                                                                                                                                                                                       
        at org.hibernate.persister.entity.AbstractEntityPersister.updateOrInsert(AbstractEntityPersister.java:3063)                                                                                                                                                                                                                                               
        at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3443)                                                                                                                                                                                                                                                       
        at org.hibernate.action.internal.EntityUpdateAction.execute(EntityUpdateAction.java:145)                                                                                                                                                                                                                                                                  
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:589)                                                                                                                                                                                                                                                                              
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:463)                                                                                                                                                                                                                                                                              
        at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:337)                                                                                                                                                                                                                                   
        at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:39)                                                                                                                                                                                                                                                      
        at org.hibernate.internal.SessionImpl.flush(SessionImpl.java:1295)                                                                                                                                                                                                                                                                                        
        at org.grails.orm.hibernate.AbstractHibernateGormInstanceApi.flushSession(AbstractHibernateGormInstanceApi.groovy:284)                                                                                                                                                                                                                                    
        at org.grails.orm.hibernate.AbstractHibernateGormInstanceApi$_performSave_closure3.doCall(AbstractHibernateGormInstanceApi.groovy:249)                                                                                                                                                                                                                    
        at org.grails.orm.hibernate.GrailsHibernateTemplate.doExecute(GrailsHibernateTemplate.java:299)                                                                                                                                                                                                                                                           
        ... 96 common frames omitted 
</details>

--- a/rundeckapp/grails-app/services/rundeck/services/DbStorageService.groovy
+++ b/rundeckapp/grails-app/services/rundeck/services/DbStorageService.groovy
@@ -23,6 +23,7 @@ import com.dtolabs.rundeck.plugins.ServiceNameConstants
 import com.dtolabs.rundeck.plugins.descriptions.PluginDescription
 import com.dtolabs.rundeck.plugins.storage.StoragePlugin
 import com.dtolabs.rundeck.server.storage.NamespacedStorage
+import grails.transaction.NotTransactional
 import grails.transaction.Transactional
 import org.rundeck.storage.api.HasInputStream
 import org.rundeck.storage.api.Path
@@ -250,12 +251,15 @@ class DbStorageService implements NamespacedStorage{
     }
 
     @Override
+    @NotTransactional
     boolean deleteResource(String ns,Path path) {
         Storage storage1 = findResource(ns,path)
         if (!storage1) {
             throw StorageException.deleteException(path, "Not found")
         }
-        storage1.delete(flush: true)
+        Storage.withNewTransaction {
+            storage1.delete(flush: true)
+        }
         return true
     }
 
@@ -264,6 +268,7 @@ class DbStorageService implements NamespacedStorage{
     }
 
     @Override
+    @NotTransactional
     Resource<ResourceMeta> createResource(String ns,Path path, ResourceMeta content) {
         if (path.path.contains('../')) {
             throw StorageException.createException(path, "Invalid path: ${path.path}")
@@ -282,36 +287,40 @@ class DbStorageService implements NamespacedStorage{
         Storage saved=null;
         def data = content.getInputStream().bytes
         def saveStorage={
-            try {
-                if (id) {
-                    storage = Storage.get(id)
-                } else {
-                    storage = new Storage()
-                }
+            Storage.withNewTransaction {
+                try {
+                    if (id) {
+                        storage = Storage.findById(id, [lock: true])
+                        storage.refresh()
+                    } else {
+                        storage = new Storage()
+                    }
 
-                storage.namespace = namespace ? namespace : null
-                storage.path = path.path
-                Map<String, String> newdata = storage.storageMeta?:[:]
-                storage.storageMeta = newdata + content.meta
-                storage.data = data
-                saved = storage.save(flush: true)
-                if (!saved) {
-                    throw new StorageException("Failed to save content at path ${path}: validation error: " +
-                            storage.errors.allErrors.collect { it.toString() }.join("; "),
-                            StorageException.Event.valueOf(event.toUpperCase()),
-                            path)
-                }
+                    storage.namespace = namespace ? namespace : null
+                    storage.path = path.path
+                    Map<String, String> newdata = storage.storageMeta?:[:]
+                    storage.storageMeta = newdata + content.meta
+                    storage.data = data
+                    saved = storage.save(flush: true)
+                    if (!saved) {
+                        throw new StorageException("Failed to save content at path ${path}: validation error: " +
+                                storage.errors.allErrors.collect { it.toString() }.join("; "),
+                                StorageException.Event.valueOf(event.toUpperCase()),
+                                path)
+                    }
 
-                retry = false
-                return true;
-            } catch (org.springframework.dao.ConcurrencyFailureException e) {
-                if (!retry) {
-                    throw new StorageException("Failed to save content at path ${path}: content was modified", e,
-                            StorageException.Event.valueOf(event.toUpperCase()),
-                            path)
-                } else {
-                    log.warn("saveStorage optimistic locking failure for ${path}, retrying...")
-                    Thread.sleep(1000)
+                    retry = false
+                    return true;
+                } catch (org.springframework.dao.ConcurrencyFailureException e) {
+                    if (!retry) {
+                        throw new StorageException("Failed to save content at path ${path}: content was modified", e,
+                                StorageException.Event.valueOf(event.toUpperCase()),
+                                path)
+                    } else {
+                        log.warn("exception catch: ${e.getClass()}")
+                        log.warn("saveStorage optimistic locking failure for ${path}, retrying...")
+                        Thread.sleep(1000)
+                    }
                 }
             }
             return false;
@@ -335,6 +344,7 @@ class DbStorageService implements NamespacedStorage{
     }
 
     @Override
+    @NotTransactional
     Resource<ResourceMeta> updateResource(String ns,Path path, ResourceMeta content) {
         if (path.path.contains('../')) {
             throw StorageException.updateException(path, "Invalid path: ${path.path}")

